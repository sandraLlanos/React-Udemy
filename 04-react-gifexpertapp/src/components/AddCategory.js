import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState(''); // ''

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value );
        console.log('handleInputChange is called');
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log('HandldSubmit is called');
        
        if ( inputValue.trim().length > 2 ) {
            console.log('setCategories is called');
            setCategories( cats => [ inputValue, ...cats, ] );
            setInputValue('');
        }

    }

    return (
        <form onSubmit={ handleSubmit }>
            <p> {inputValue} </p>
            <input 
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
