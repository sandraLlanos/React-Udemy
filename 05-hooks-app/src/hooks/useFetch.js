import { useEffect, useState, useRef } from "react"

export const useFetch = ( url ) => {
    
    const [state, setState] = useState({data:null, loading:true, error:null});

    const isMounted = useRef(true)   

    useEffect(() => {
        return () => {
            isMounted.current = false;
        }
    }, [])

    useEffect(() => {
        setState({
            data: null,
            loading:true, 
            error:null
        });
        fetch(url)
            .then(resp => resp.json())
            .then(data => {
                setTimeout(() => {
                    if(isMounted.current){
                        setState({
                        data,
                            loading:false, 
                            error:null
                        })
                    }
                    else {
                        console.log('unmonunt component');
                    }
                }, 4000);
            })
    }, [url]);

    return state;
}
