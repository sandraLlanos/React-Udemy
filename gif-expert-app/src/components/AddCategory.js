import React, { useState } from 'react'

const AddCategory = () => {
    const [inputValue, setInputValue] = useState('hello word!');
    const handleInputValue = (e) => {
        setInputValue(e.target.value);
    } 
    const handleSubmit = (e) => {
        e.preventDefault();
        console.log('Form submited');
    }
    return (
        <form onSubmit={handleSubmit}>
            <input 
                type="text"
                value={inputValue}
                onChange={handleInputValue}
            />
        </form>
    )
}

export default AddCategory
