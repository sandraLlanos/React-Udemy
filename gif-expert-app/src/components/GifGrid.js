import React from 'react'

export const GifGrid = ( {category } ) => {

    const getGifs = async() => {
        const apiUrl = 'https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&limit=10&api_key=XRnsOEVygIHs5bH4hMpOall24oypnnmH';

        const resp = await fetch( apiUrl );
        const {data} = await resp.json();
        // console.log(data[0].id)
        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images.downsized_medium.url
            }
        })
        console.log(gifs);
    }

    getGifs();

    return (
        <>
            <h3> {category} </h3>
            
        </>
    )
}
