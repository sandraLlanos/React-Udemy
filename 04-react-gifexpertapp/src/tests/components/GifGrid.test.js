import React from 'react';
import { GifGrid } from "../../components/GifGrid";
import { shallow } from "enzyme";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('test in <GifGrid /> component', () => {
    
    const category = 'One Punch';

    test('should load successfully', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot();
    });
    test('should show items when the images has been loaded useFetchGifs', () => {
        
        const gifs = [{
            id: 'ABC',
            title: 'anything',
            url: 'https://localhost/anything.jpg'
        }]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow( <GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot();

    });

});