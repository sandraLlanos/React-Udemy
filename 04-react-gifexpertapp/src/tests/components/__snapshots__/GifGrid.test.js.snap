// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <GifGrid /> component should load successfully 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`test in <GifGrid /> component should show items when the images has been loaded useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="anything"
      url="https://localhost/anything.jpg"
    />
  </div>
</Fragment>
`;
